libs/go/aealite/protocols/dialogue.go

```diff
@@ -0,0 +1,305 @@
+package protocols
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+)
+
+type Address string
+type Performative []string
+type SomeMessageType string
+
+const (
+	NONCE_BYTES_NB = 32
+)
+
+// type Dialogue struct {
+// 	_dialogues_storage
+// 	_dialogues_stats
+// 	_keep_terminal_state_dialogues
+// 	_message_class
+// 	_dialogue_class
+// 	_role_from_first_message
+// 	_self_address
+// 	_outgoing_messages
+// 	_incoming_messages
+// }
+
+// TODO
+// Define RoleType, RuleType and figure out how to find role
+type RoleType Address
+type RuleType Address
+
+type MessageId int
+type targetMessage InitialMessage
+type target []int
+type DialogueInterface interface {
+	initialize(dialogueLabel Label, message InitialMessage, selfAddress Address, role RoleType)
+	equal(Dialogue) bool // compare two dialogues
+	getDialogueLabel() Label
+	getIncompleteDIalgueLabel() Label
+	getDialogueLabels() []Label // returns both dialogue and incomplete dialog labels Label
+	getSelfAddress() Address
+	getRole() RoleType
+	getRules() RuleType
+	getMessageCLass() InitialMessage
+	isSelfInitiated() bool
+	getLastIncomingMessage() InitialMessage
+	getLastOutgoingMessage() InitialMessage
```

Comment:
type..

Reply:
just call it `AbstractMessage` and define an interface for it according to what we do on python side